### PR TITLE
Enh: add glossary

### PR DESCRIPTION
**Pull request by lwasser**
@kierisi it turns out that a glossary isn't an additional infra it's just a directive. 

To make sure terms highlight and link to the glossary you can use this syntax:

How you call the glossary:

{term}`sdist`

i had thought it might automagically pick up the terms each time we add one but that is not the case. we will have to add references to the thing defined in the glossary! but this should get us started! and you are welcome to submit a fresh PR if adding to this one feels like more work!! i didn't really do much here but figure out how work :) 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206399753784684

### CODE REVIEW

File: glossary.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Glossary
+
+A glossary of common terms used throughout Jupyter Book.
+
+```{glossary}
+
+[sdist](#python-source-distribution)
```
Comment (lwasser) [inline glossary.md:7]: All terms need to be lined up like this. you can define multiple "terms" with a single definition.

File: tutorials/1-installable-code.md [hunk 2]
```diff
@@ -3,6 +3,9 @@
 :og:title: Make your Python code installable so it can be used across projects
 ---
 
+How you call the glossary:
+{term}`sdist`
```
Comment (lwasser) [inline tutorials/1-installable-code.md:7]: This is how you call a glossary term in the text using myst

File: glossary.md (hunk 1 above)
Comment (kierisi) [reply glossary.md:7]: @lwasser this is great, thank you so much! I'll start adding terms to this as I come across them and we can go from there!

**Comment by willingc** [comment]
Doing a monthly review of PRs, should we close or finish off this PR? @lwasser

**Comment by lwasser** [comment]
@willingc thank you!! i'll close it. i left a link to this as an example in the open issue #141 but we can still close this issue! ✨  thank you for doing this monthly issue round up!!

**Pull request closed without merging**